### PR TITLE
Enable Audio feature.

### PR DESCRIPTION
**Pull request by Sunwoo-Nam**
1. Connect with TCP socket instead of SyncSocket.
2. Sync a shared memory for audio data from RP to BP.

### CODE REVIEW

**Comment by Sunwoo-Nam** [comment]
I'm refactoring this PR. Please do not review until update.

**Comment by Sunwoo-Nam** [comment]
Now, the renderer process creates a TCP server and propagates the ip address and port number to the browser process for TCP connection.

File: mojo/public/cpp/platform/tcp_platform_handle_utils_posix.cc [hunk 1]
```diff
@@ -168,4 +169,23 @@ bool TCPServerAcceptConnection(const base::PlatformFile server_socket,
 #endif  // defined(OS_NACL)
 }
 
+bool GetSelfIPAddress(std::string* address) {
+  struct hostent* host_entry;
+  char hostname[80];
+  gethostname(hostname, sizeof(hostname));
+  host_entry = gethostbyname(hostname);
```
Comment (younghajung) [inline mojo/public/cpp/platform/tcp_platform_handle_utils_posix.cc]: In my local machine, `gethostbyname` just returns 127.0.1.1.
I think `gethostbyname` is based on hosts file.
Hence, if there is no information about own IP address in hosts file, we cannot get own address.
How about using `getifaddrs`?

Comment (Sunwoo-Nam) [reply mojo/public/cpp/platform/tcp_platform_handle_utils_posix.cc]: I think It's better to remove this code :)
I am going to update PR. Thank you.

File: content/renderer/media/audio/mojo_audio_output_ipc.cc [hunk 2]
```diff
@@ -233,6 +237,17 @@ void MojoAudioOutputIPC::Created(
       data_pipe->shared_memory;
   DCHECK(shared_memory_region.IsValid());
 
+#if defined(CASTANETS)
+  // If socket_handle is invalid.
+  if (socket_handle < 0) {
+    shared_memory_region_ = std::move(shared_memory_region);
```
Comment (is46-kim) [inline content/renderer/media/audio/mojo_audio_output_ipc.cc]: How about passing |shared_memory_region| with parameter to callback function without |shared_memory_region_| member?

Comment (Sunwoo-Nam) [reply content/renderer/media/audio/mojo_audio_output_ipc.cc]: Thank you for your point.

**Review by hyowon — APPROVED** [review]
It would be better if the audio service does not directly control the connection of SyncSocket. But let's merge this PR as audio works well with this implementation.

**Pull request merged**